src/components/Banner/Banner.tsx

```diff
@@ -55,202 +59,210 @@ export interface BannerProps {
   stopAnnouncements?: boolean;
 }
 
-export class Banner extends PureComponent<BannerProps, State> {
-  static contextType = FeaturesContext;
-  context!: React.ContextType<typeof FeaturesContext>;
-
-  state: State = {
-    showFocus: false,
-  };
+interface BannerAttributes {
+  iconColor: IconProps['color'];
+  defaultIcon: IconProps['source'];
+  ariaRoleType?: 'alert';
+}
 
-  private wrapper = createRef<HTMLDivElement>();
+function useBannerAttributes(status: BannerProps['status']): BannerAttributes {
```

Comment:
Nitpick: The first function in our component files tends to be the component function itself, so having `useBannerAttributes`, and `useBannerFocus` above `Banner` means you have to scroll half the file to see the most important function, which feels weird. Can we have Banner be the first function, followed by useBannerAttributes/useBannerFocus?

Reply:
Sure thing! I’m not a big fan of relying on hoisting, but I see that’s something we do a lot at Shopify! Maybe I’ll start liking it at some point 😄

Reply:
Right, interesting comparison with classes. 

> Maybe I’ll start liking it at some point 😄

I think I’ve already started liking it tbh. Until recently, I used arrow functions everywhere, so this was out of the question. Now that I tend to default to function declarations, my brain probably needs to adjust to this being possible 😄 

I think somehow I don’t (didn’t) like the idea of using something that you… haven’t seen before? But it’s kind of a stupid way to think about it.

Among other things, functions allow you to hide “implementation details”, and putting “details” at the bottom makes sense since… they’re less important 😄 
They’re in the same file too, so jumping to them when needed is not a big hassle.

There‘s definitely value in knowing that the first function definition will always be the component, so you can skim through files quickly.